app/scripts/components/exploration/types.d.ts.ts

```diff
@@ -107,6 +109,26 @@ export interface TimelineDatasetSettings {
 // Tile endpoints for the layer given the current map view.
 type TimelineDatasetMeta = Record<string, any>;
 
+// Holds only dataset needed for visualization (Subset of timeline dataset)
+// @ TODO: Rename Timeline specific variable names
+export interface VizDatasetIDLE {
+  status: TimelineDatasetStatus.IDLE | TimelineDatasetStatus.LOADING;
+  data: EnhancedDatasetLayer;
+  error: null;
+  settings: TimelineDatasetSettings;
+  meta?: TimelineDatasetMeta;
+}
+
+export interface VizDatasetSUCCESS {
```

Comment:
This is still a rough sketch of the new type only for visualization. But this is the general direction that I am trying to go.

Reply:
Thanks, this aligns with what I also imagined during our call on Friday!